File: services/src/northbound/src/test/java/org/bitbucket/openkilda/northbound/controller/TestConfig.java
/* Copyright 2017 Telstra Open Source
 *
 *   Licensed under the Apache License, Version 2.0 (the "License");
 *   you may not use this file except in compliance with the License.
 *   You may obtain a copy of the License at
 *
 *       http://www.apache.org/licenses/LICENSE-2.0
 *
 *   Unless required by applicable law or agreed to in writing, software
 *   distributed under the License is distributed on an "AS IS" BASIS,
 *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *   See the License for the specific language governing permissions and
 *   limitations under the License.
 */

package org.bitbucket.openkilda.northbound.controller;

import org.bitbucket.openkilda.northbound.config.SecurityConfig;
import org.bitbucket.openkilda.northbound.config.WebConfig;
import org.bitbucket.openkilda.northbound.messaging.HealthCheckMessageConsumer;
import org.bitbucket.openkilda.northbound.messaging.MessageConsumer;
import org.bitbucket.openkilda.northbound.messaging.MessageProducer;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.ComponentScan;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Import;
import org.springframework.context.annotation.PropertySource;
import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;

import java.util.Map;

/**
 * The Test configuration.
 */
@Configuration
@EnableWebSecurity
@Import({WebConfig.class, SecurityConfig.class})
@ComponentScan({
        "org.bitbucket.openkilda.northbound.controller",
        "org.bitbucket.openkilda.northbound.model",
        "org.bitbucket.openkilda.northbound.service",
        "org.bitbucket.openkilda.northbound.utils"})
@PropertySource({"classpath:northbound.properties"})
public class TestConfig {
    @Bean
    public MessageConsumer messageConsumer() {
        return new TestMessageMock();
    }

    @Bean
    public MessageProducer messageProducer() {
        return new TestMessageMock();
    }

    @Bean
    public HealthCheckMessageConsumer healthCheckMessageConsumer() {
        return new TestHealthCheckMessageMock();
    }

    private class TestHealthCheckMessageMock implements HealthCheckMessageConsumer {

        @Override
        public Map<String, String> poll(String correlationId) {
            return null;
        }

        @Override
        public void clear() {

        }
    }
}

File: services/topology-engine/queue-engine/topologylistener/tests/test_config.py
# Copyright 2017 Telstra Open Source
#
#   Licensed under the Apache License, Version 2.0 (the "License");
#   you may not use this file except in compliance with the License.
#   You may obtain a copy of the License at
#
#       http://www.apache.org/licenses/LICENSE-2.0
#
#   Unless required by applicable law or agreed to in writing, software
#   distributed under the License is distributed on an "AS IS" BASIS,
#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
#   See the License for the specific language governing permissions and
#   limitations under the License.
#

from topologylistener import db
from topologylistener import messageclasses
from topologylistener.tests import share


def make_feature_toggle_request(payload):
    data = payload.copy()
    data['clazz'] = (
        'org.openkilda.messaging.command.system.FeatureToggleRequest')
    message = share.command(data)
    return messageclasses.MessageItem(message).handle()


def clean_neo4j_test_data(tx):
    tx.run('MATCH (c:config) DETACH DELETE c')


class TestConfig(share.AbstractTest):
    def setUp(self):
        super(TestConfig, self).setUp()
        with share.env.neo4j_connect.begin() as tx:
            clean_neo4j_test_data(tx)
        messageclasses.read_config()

    def test_update(self):
        features_request = share.feature_toggle_request(**{
            messageclasses.features_status_app_to_transport_map[
                messageclasses.FEATURE_REROUTE_ON_ISL_DISCOVERY]: False})
        self.assertTrue(share.feed_message(share.command(features_request)))

        with share.env.neo4j_connect.begin() as tx:
            cursor = tx.run(
                    'MATCH (c:config {name: "config"}) RETURN c LIMIT 2')
            db_config = db.fetch_one(cursor)['c']
            self.assertEqual(
                    False,
                    db_config[
                        messageclasses.features_status_app_to_transport_map[
                            messageclasses.FEATURE_REROUTE_ON_ISL_DISCOVERY]])

    def test_existing(self):
        with share.env.neo4j_connect.begin() as tx:
            q = 'CREATE (c:config) SET c = $data'
            p = {
                'data': {
                    'name': 'config',
                    'reroute_on_isl_discovery': False,
                    'unpush_flow': False,
                    'sync_rules': False,
                    'push_flow': False,
                    'delete_flow': False,
                    'create_flow': False,
                    'update_flow': False}}
            tx.run(q, p)

        messageclasses.read_config()
        self.assertFalse(
                messageclasses.features_status[
                    messageclasses.FEATURE_REROUTE_ON_ISL_DISCOVERY])
